storage/bucket_test.go

```diff
@@ -1210,3 +1212,301 @@ func TestDetectDefaultGoogleAccessID(t *testing.T) {
 		})
 	}
 }
+
+// TestBucketSignedURL_Endpoint_Emulator_Host tests that Bucket.SignedURl
+// respects the host set in STORAGE_EMULATOR_HOST and/or in option.WithEndpoint
+func TestBucketSignedURL_Endpoint_Emulator_Host(t *testing.T) {
+	expires, _ := time.Parse(time.RFC3339, "2002-10-02T10:00:00-05:00")
+	bucketName := "bucket-name"
+	objectName := "obj-name"
+
+	localhost9000 := "localhost:9000"
+	localhost6000Https := "https://localhost:6000"
+
+	tests := []struct {
+		desc         string
+		emulatorHost string
+		endpoint     *string
+		now          time.Time
+		opts         *SignedURLOptions
+		// Note for future implementors: X-Goog-Signature generated by having
+		// the client run through its algorithm with pre-defined input and copy
+		// pasting the output. These tests are not great for testing whether
+		// the right signature is calculated - instead we rely on the backend
+		// and integration tests for that.
+		want string
+	}{
+		{
+			desc:         "SignURLV4 creates link to resources in emulator",
```

Comment:
These cases really seem like they belong in the conformance tests for signed URLs once the project is complete. Could you add a TODO to move them? (and I guess this would also require exporting the hostname option rather than having it be internal only)